### ISSUE TITLE
Storefront Hot Proxy - CORS error

### ISSUE THREAD

**Issue by shopwareBot** [shopware/platform#4094]
### Description

For a product with variants, these cannot be accessed in frontend, as a variant selection causes an Ajax CORS error. This is due the URL not including the watcher port.

```
Access to XMLHttpRequest at 'http://localhost/detail/...' from origin 'http://localhost:9998' has been blocked by CORS policy: Request header field x-requested-with is not allowed by Access-Control-Allow-Headers in preflight response.
```

### Environment

Local DEV environment, all latest versions as of 2023-11.

- Shopware 6.5
- Zenit Theme 3.5.5
- Docker + Dockware (MacOS 14)
- Google Chrome 119

### Steps to reproduce

- 1. Create a product with variants to select from
- 2. Start the frontend watcher (Shopware native or Dockware command, should not matter)
- 3. Navigate to the product detail page
- 4. Open the browser console
- 5. Click a variant

### Expected result

- Ajax request passes, page is reloaded with variant content.

### Actual result

- Ajax request fails with CORS.

### Workaround

- Rewriting the URL with a proxy tool like 'Proxyman' will lead to the expected result. This should oprove, that it's not environment specific or custom code at fault.

### Attachments

- The demo shows another domain than localhost (proxyman.debug), which only fails, if the proxy is disabled. I've just chosen to use it, so I can switch it faster as workaround (localhost does not work with the proxy tool).

### Related tickets

- NEXT-10750
  [CORS.png|https://sbp-issuetracker-attachments.s3.eu-west-1.amazonaws.com/1699536538_654cde9a42dae_1699536536_654cde9860390_CORS.png]

**Comment by shopwareBot** [comment]
[public] Automated response: This issue is linked to the internal issue [https://shopware.atlassian.net/browse/NEXT-31611](https://shopware.atlassian.net/browse/NEXT-31611).

 

[created from NEXT-31611, comment 485495]

**Comment by shopwareBot** [comment]
[public] Hello,

 Within our company roadmap and work capacity, we try to address each bug or improvement request but admit that not each one will be resolved. To continue our culture of honesty and openness, we are closing this issue to focus on our roadmap on behalf of all Shopware users.

 If you feel like this issue is still impacting you, please create a new issue and let us know. Thank you again for your valuable feedback.



[created from NEXT-31611, comment 492281]
